src/libraries/Common/src/Extensions/Logging/NullExternalScopeProvider.cs

```diff
@@ -20,12 +22,12 @@ private NullExternalScopeProvider()
         public static IExternalScopeProvider Instance { get; } = new NullExternalScopeProvider();
 
         /// <inheritdoc />
-        void IExternalScopeProvider.ForEachScope<TState>(Action<object, TState> callback, TState state)
+        void IExternalScopeProvider.ForEachScope<TState>(Action<object?, TState> callback, TState state)
```

Comment:
should TState be nullable?

Reply:
> should TState be nullable?

It looks fine as is.  If a developer wants to provide a nullable TState, they can.  I don't see a benefit here in constraining it to be notnull.